### ISSUE TITLE
Full executable paths are stripped on format

### ISSUE THREAD

**Issue by kroppt** [PowerShell/vscode-powershell#2130]
### System Details

System Details Output
```
> $PSVersionTable
Name                           Value
----                           -----
PSVersion                      5.1.18362.145
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.18362.145
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```
Codium version
```
> codium -v
1.37.0
036a6b1d3ac84e5ca96a17a44e63a87971f8fcc8
x64
```
Extension version
```
> codium  --list-extensions --show-versions
...
ms-vscode.powershell@2019.5.0
```
Editor Services version
```
> $psEditor.EditorServicesVersion

Major  Minor  Build  Revision
-----  -----  -----  --------
1      12     1      0
```

### Issue Description

This extension is changing lines for me which I do not think should be changed. And the criteria doesn't make sense. If the full path used exists on the file system, then the path is stripped down to the last element.

#### Expected Behaviour

Full path to an executable is used in a PowerShell script, and the formatter leaves it be.

#### Actual Behaviour

Full path to an executable is used in a PowerShell script, and the formatter cuts it down to the last element.

### Attached Logs

[vscodium-PSextension-logs.zip](https://github.com/PowerShell/vscode-powershell/files/3486708/vscodium-PSextension-logs.zip)

### Postscript

I think that having an option to disable this functionality would be good. I could not find the rule associated with this formatting in PSScriptAnalyzer.

Along with this, when enabled, full executable paths should always be stripped, not just those whose paths exist on the system.

**Comment by SydneyhSmith** [comment]
@kroppt thanks for opening this issue and providing logs...it would be super helpful if you also provided repro-steps of a path that is getting stripped down, or a screenshot/gif of this issue

**Comment by kroppt** [comment]
1. Make file somewhere (e.g. `C:\test\test.exe`)
2. Enter full file path, with or without drive letter (e.g. `C:\test\test.exe`
3. Format Document (Shift-Alt-F)

A side note: in testing myself, I needed to put the file under its own folder for the drive-letter-less path to work, i.e. `\test.exe` did not change in format.
Additionally, you can give it the path to the folder, and it will do the same.

**Comment by kroppt** [comment]
It's not inactive, actually. It still occurs.

**Comment by rkeithhill** [comment]
Formatting is provided by the PSScriptAnalyzer project.  @SydneyhSmith Can you move this or perhaps mark it as duplicate if there's already a PSSA bug on this?

**Comment by kroppt** [comment]
When I format the line, it get changed as described, but when I look at rule violations using `Invoke-ScriptAnalyzer -ReportSummary .\test.ps1`, none appear. I also could not find a relevant rules.

**Comment by TylerLeonhardt** [comment]
We call `Invoke-Formatter` for this scenario.

**Comment by kroppt** [comment]
> We call `Invoke-Formatter -ScriptDefinition <your script> -Settings <the settings>` for this scenario.

It's not possible.
When I install PSScriptAnalyzer through Chocolatey, the issue in VSCode goes away, and it is no longer replicable. I guess the default version that comes with the extension has a bug.

> Can you try this? Also are there any formatting-related settings you have turned on?

I do have some settings:
```
"powershell.codeFormatting.useCorrectCasing": true,
"powershell.codeFormatting.newLineAfterCloseBrace": false,
```

**Comment by TylerLeonhardt** [comment]
What version of PSScriptAnalyzer did you install? 1.18.1? Currently the extension ships 1.18.0 because we've been heads down on some other work but if upgrading PSScriptAnalyzer has fixed your issue, then it was probably fixed in 1.18.1

**Comment by kroppt** [comment]
Yes, I installed 1.18.1 to fix it.

**Comment by TylerLeonhardt** [comment]
Ok great - I'll close this issue since it was related to PSScriptAnalyzer and is fixed in 1.18.1. We should be doing a release of the vscode extension in the next couple weeks which will include a new PSSA.
